### PR TITLE
Improve error message initialisation SpanROI

### PR DESCRIPTION
**Pull request by ericpre**
### Progress of the PR
- [x] Improve error message when initialising ROI with left >= right.
- [x] add tests,
- [x] update changelog, 
- [x] ready for review.

### Minimal example of the bug fix or the new feature
```python
import hyperspy.api as hs
hs.roi.SpanROI(30, 15)
```
raises an confusing message: 
```python
Traceback (most recent call last):

  File "<ipython-input-1-1e344269bc12>", line 2, in <module>
    hs.roi.SpanROI(30, 15)

  File "/home/eric/Dev/hyperspy/hyperspy/roi.py", line 688, in __init__
    self.left, self.right = left, right

  File "/opt/miniconda3/lib/python3.8/site-packages/traits/trait_notifiers.py", line 346, in __call__
    handle_exception(object, trait_name, old, new)

  File "/opt/miniconda3/lib/python3.8/site-packages/traits/trait_notifiers.py", line 148, in _handle_exception
    raise excp

  File "/opt/miniconda3/lib/python3.8/site-packages/traits/trait_notifiers.py", line 340, in __call__
    self.handler(*args)

  File "/home/eric/Dev/hyperspy/hyperspy/roi.py", line 703, in _right_changed
    self.right = old

  File "/opt/miniconda3/lib/python3.8/site-packages/traits/trait_notifiers.py", line 346, in __call__
    handle_exception(object, trait_name, old, new)

  File "/opt/miniconda3/lib/python3.8/site-packages/traits/trait_notifiers.py", line 148, in _handle_exception
    raise excp

  File "/opt/miniconda3/lib/python3.8/site-packages/traits/trait_notifiers.py", line 340, in __call__
    self.handler(*args)

  File "/home/eric/Dev/hyperspy/hyperspy/roi.py", line 702, in _right_changed
    self.left is not t.Undefined and new <= self.left:

TypeError: '<=' not supported between instances of '_Undefined' and 'float'
```
With this PR:
```python
Traceback (most recent call last):

  File "<ipython-input-1-1e344269bc12>", line 2, in <module>
    hs.roi.SpanROI(30, 15)

  File "/home/eric/Dev/hyperspy/hyperspy/roi.py", line 689, in __init__
    raise ValueError("`left` must be smaller than `right`.")

ValueError: `left` must be smaller than `right`.
```

### CODE REVIEW

**Comment by ericpre** [comment]
@magnunor, @thomasaarholt, this PR is simple, can one of you two please review it?

File: hyperspy/roi.py [hunk 1]
```diff
@@ -702,16 +702,16 @@ class SpanROI(BaseInteractiveROI):
     _ndim = 1
 
     def __init__(self, left, right):
-        super(SpanROI, self).__init__()
+        super().__init__()
         self._bounds_check = True   # Use reponsibly!
```
Comment (magnunor) [inline hyperspy/roi.py]: responsibly

File: hyperspy/roi.py [hunk 2]
```diff
@@ -702,16 +702,16 @@ class SpanROI(BaseInteractiveROI):
...
+        if left >= right:
+            raise ValueError("`left` must be smaller than `right`.")
```
Comment (magnunor) [inline hyperspy/roi.py]: Could maybe add the values themselves to the error message. Sometimes function which raises the error isn't the one which the user called, so having the values themselves could be useful in that case. For example:

```python
raise ValueError("`left` ({0}) must be smaller than `right` ({1}).".format(left, right))
```

**Review by magnunor — COMMENTED** [review]
Nice improvement! Good error messages are important.

Some comments in the code.

**Comment by ericpre** [comment]
All done with a a few deprecation warning fixes.

**Pull request merged**